mosip-identity-plugin/src/main/java/io/mosip/signup/plugin/mosipid/service/IdrepoProfileRegistryPluginImpl.java

```diff
@@ -160,15 +168,186 @@ public void init() {
+                result.set(name, nameNode);
+            }
+
+            // Iterate through each field value and add to the nested structure
+            for (JsonNode fv : fieldValues) {
+                String code = fv.hasNonNull("code") ? fv.get("code").asText() : null;
+                String value = fv.hasNonNull("value") ? fv.get("value").asText() : null;
+                if (code == null || value == null) continue;
+
+                ObjectNode langMap = (ObjectNode) nameNode.get(code);
+                if (langMap == null) {
+                    langMap = objectMapper.createObjectNode();
+                    nameNode.set(code, langMap);
+                }
+                langMap.put(lang, value);
+            }
+        }
+    }
+
+    /**
+     * Processes the document categories JSON list and adds their structured data into the provided ObjectNode.
+     * @param documentCategories List of JSON nodes representing document categories
+     * @param result The ObjectNode where data is accumulated
+     */
+    private void processDocumentCategoriesAndTypes(List<JsonNode> documentCategories, ObjectNode result) {
+        for (JsonNode item : documentCategories) {
+            String categoryCode = item.hasNonNull("code") ? item.get("code").asText() : null;
+            String langCode = item.hasNonNull("langCode") ? item.get("langCode").asText() : null;
+            JsonNode documentTypes = item.get("documentTypes");
+
+            // Skip if required fields are missing or documentTypes is not an array
+            if (categoryCode == null || langCode == null || documentTypes == null || !documentTypes.isArray())
+                continue;
+
+            // Get or create the node for the document category code
+            ObjectNode docTypeMap = (ObjectNode) result.get(categoryCode);
+            if (docTypeMap == null) {
+                docTypeMap = objectMapper.createObjectNode();
+                result.set(categoryCode, docTypeMap);
+            }
+
+            // Iterate through each document type and add to the nested structure
+            for (JsonNode docType : documentTypes) {
+                String docTypeCode = docType.hasNonNull("code") ? docType.get("code").asText() : null;
+                String docTypeName = docType.hasNonNull("name") ? docType.get("name").asText() : null;
+                if (docTypeCode == null || docTypeName == null) continue;
+
+                ObjectNode langMap = (ObjectNode) docTypeMap.get(docTypeCode);
+                if (langMap == null) {
+                    langMap = objectMapper.createObjectNode();
+                    docTypeMap.set(docTypeCode, langMap);
+                }
+                langMap.put(langCode, docTypeName);
+            }
+        }
+    }
+
+    public String buildDynamicFieldsUrl(int pageNumber, int pageSize) {
+        return String.format("%s?pageNumber=%d&pageSize=%d", dynamicFieldsBaseUrl, pageNumber, pageSize);
```

Comment:
why is this method "public"? and instead of hardcoding the URL path, can we take it as part of dynamicFieldsBaseUrl?